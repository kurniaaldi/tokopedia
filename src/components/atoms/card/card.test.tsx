import { render, screen } from "@testing-library/react";
import Card from "./index";
import renderer from "react-test-renderer";

test("card", () => {
  const Component = () => <p>card</p>;
  render(
    <Card>
      <Component />
    </Card>
  );

  const cardElement = screen.getByTestId("card-id");
  expect(cardElement).toBeInTheDocument();
  expect(cardElement).toHaveTextContent("card");
  expect(cardElement).toContainHTML("<p>card</p>");
});

test("matches snapshot", () => {
  const Component = () => <p>card</p>;

  const tree = renderer
    .create(
      <Card>
        <Component />
      </Card>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
