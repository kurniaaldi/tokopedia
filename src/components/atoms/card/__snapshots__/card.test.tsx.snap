// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-wg4ovg"
  data-testid="card-id"
>
  <p>
    card
  </p>
</div>
`;
